banyand/trace/block_metadata.go

```diff
@@ -0,0 +1,290 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package trace
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/apache/skywalking-banyandb/pkg/convert"
+	"github.com/apache/skywalking-banyandb/pkg/encoding"
+	pbv1 "github.com/apache/skywalking-banyandb/pkg/pb/v1"
+	"github.com/apache/skywalking-banyandb/pkg/pool"
+	"github.com/apache/skywalking-banyandb/pkg/query/model"
+)
+
+type dataBlock struct {
+	offset uint64
+	size   uint64
+}
+
+func (d *dataBlock) reset() {
+	d.offset = 0
+	d.size = 0
+}
+
+func (d *dataBlock) copyFrom(src *dataBlock) {
+	d.offset = src.offset
+	d.size = src.size
+}
+
+func (d *dataBlock) marshal(dst []byte) []byte {
+	dst = encoding.VarUint64ToBytes(dst, d.offset)
+	dst = encoding.VarUint64ToBytes(dst, d.size)
+	return dst
+}
+
+func (d *dataBlock) unmarshal(src []byte) []byte {
+	src, n := encoding.BytesToVarUint64(src)
+	d.offset = n
+
+	src, n = encoding.BytesToVarUint64(src)
+	d.size = n
+	return src
+}
+
+type blockMetadata struct {
+	tags                      map[string]*dataBlock
+	tagType                   map[string]pbv1.ValueType
+	spans                     *dataBlock
+	tagProjection             *model.TagProjection
+	traceID                   string
```

Comment:
The spans in a block could come from different traces. It can not contain a single traceID here.

Reply:
it could be the maxTracID and minTraceID